File: WallsForTeens/app/src/main/java/spiral/bit/dev/wallsforteens/MainActivity.java
package spiral.bit.dev.wallsforteens;

import android.content.Intent;
import android.content.SharedPreferences;
import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;
import com.google.android.material.tabs.TabLayout;
import spiral.bit.dev.wallsforteens.Adapters.TabAccessorAdapter;

@RequiresApi(api = Build.VERSION_CODES.M)
public class MainActivity extends AppCompatActivity {

    private ViewPager viewPager;
    private TabLayout tabLayout;
    private TabAccessorAdapter tabAccessorAdapter;
    SharedPreferences pref;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        pref = this.getSharedPreferences("rewards", 0);
        tabLayout = findViewById(R.id.main_tab);
        tabLayout.post(new Runnable() {
            @Override
            public void run() {
                tabLayout.setupWithViewPager(viewPager);
            }
        });
//        tabLayout.getTabAt(0).setText("Скейтбординг");
//        tabLayout.getTabAt(1).setText("Япония");
//        tabLayout.getTabAt(2).setText("Обои для девочек");
//        tabLayout.getTabAt(3).setText("Знаменитости");
//        tabLayout.getTabAt(4).setText("Обои/арты");
        tabAccessorAdapter = new TabAccessorAdapter(getSupportFragmentManager());
        viewPager = findViewById(R.id.main_tabs_pager);
        viewPager.setAdapter(tabAccessorAdapter);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case R.id.action_diamond:
                Toast.makeText(getApplicationContext(), "На вашем счету " + pref.getInt("diamonds", 0) + " алмазов.",
                        Toast.LENGTH_LONG).show();
                return true;
            case R.id.action_star:
                final String appPackageName = getPackageName();
                try {
                    startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + appPackageName)));
                } catch (android.content.ActivityNotFoundException anfe) {
                    startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://play.google.com/store/apps/details?id=" + appPackageName)));
                }
                return true;
        }
        return true;
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.main_menu, menu);
        return super.onCreateOptionsMenu(menu);
    }
}

File: MediaStoreLearn/app/src/main/java/hardcoder/dev/mediastorelearn/ui/main/MainActivity.kt
package hardcoder.dev.mediastorelearn.ui.main

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import hardcoder.dev.mediastorelearn.R
import hardcoder.dev.mediastorelearn.adapters.MenuThemeAdapter
import hardcoder.dev.mediastorelearn.ui.audio.AudioActivity
import hardcoder.dev.mediastorelearn.ui.grantPermissions.GrantPermissionsActivity
import hardcoder.dev.mediastorelearn.ui.images.ImagesActivity
import hardcoder.dev.mediastorelearn.ui.video.VideoActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private val adapter = MenuThemeAdapter {
        startActivity(
            Intent(
                this, when (it.id) {
                    0 -> GrantPermissionsActivity::class.java
                    1 -> VideoActivity::class.java
                    2 -> AudioActivity::class.java
                    3 -> ImagesActivity::class.java
                    else -> GrantPermissionsActivity::class.java
                }
            )
        )
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setUpRecyclerView()
    }

    private fun setUpRecyclerView() {
        adapter.submitList(MenuThemeStorage.items)
        menuThemeRecyclerView.adapter = adapter
    }
}